File: planning_meal/src/main/java/in/stackroute/planning_meal/model/recipesToFrontEnd/MealPlanPerDay.java
package in.stackroute.planning_meal.model.recipesToFrontEnd;

import in.stackroute.planning_meal.model.recipesToFrontEnd.MealPlanEach;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;

import org.springframework.data.mongodb.core.mapping.Document;


import java.util.List;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "mealPlan")
public class MealPlanPerDay {
    @Id
    private org.bson.types.ObjectId id;
    private int userId;
    private List<MealPlanEach> mealPlanEachList;
    private double totalCalories;
}

File: product-webapp-service/webapp/src/app/model/mealPlanPerDay.ts
import {  MealPlanEach } from "./mealPlanEach";


export class MealPlanPerDay {
    id:any;
    userId:number;
    mealPlanEachList:MealPlanEach[];
    totalCalories:number;
    constructor(
      id:any="",
      userId:number=0,
      mealPlanEachList:MealPlanEach[]=[],
      totalCalories:number=0
    ) {
      this.id=id;
      this.userId=userId;
      this.mealPlanEachList=mealPlanEachList;
      this.totalCalories=totalCalories;
    }
  }